wordpress-mobile/WordPress-Android
# [Reader] Move announcement card inside the feeds

Fixes #20621 

This improves the current behavior by moving the Reader Announcement card inside the feeds that show on the main Reader screen, instead of rendering it on the `ReaderFragment` app bar.

I was not sure if the announcement should be also included in the new `Your Tags` feed so I didn't include it in this PR, but @develric and @osullivanchris let me know if it should be also added there.

-----

## To Test:
This relies on a shared pref that is set when the card is dismissed, so to test multiple times you will need to clear app data (or if using an emulator push a shared pref file without the dismiss pref to override the app prefs file).

1. Open Jetpack
2. Turn off the `reader_announcement_card` feature config (in Debug settings)
3. Restart the app
4. Go to Reader
5. **Verify** the announcement card does not show in any feeds
6. Turn on the `reader_announcement_card` feature config (in Debug settings)
7. Restart the app
8. Go to Reader
9. **Verify** the announcement card is shown in all feeds except `Your Tags` and non-main feeds (e.g.: tag preview, blog preview)

-----

## Regression Notes

1. Potential unintended areas of impact

    - Feed behavior in "Main reader"
    - Feed behavior in other Activities

11. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests
    - Unit tests

12. What automated tests I added (or what prevented me from doing so)

    - Updated tests for existing scenarios
    - Added new tests for new scenarios

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
